pkg/images/defaults/flavor.go

```diff
@@ -102,13 +102,13 @@ type ImageFlavor struct {
 func DevelopmentBuildImageFlavor() ImageFlavor {
 	v := version.GetAllVersionsDevelopment()
 	return ImageFlavor{
-		MainRegistry:       "docker.io/stackrox",
+		MainRegistry:       "quay.io/stackrox",
```

Comment:
There's no such thing as `quay.io/stackrox`. Technically, there _used_ to be `quay.io/stackrox` but that was used for one specific purpose. I don't think we use it any more now.

Please take a look at the stage where we push images.
https://github.com/stackrox/stackrox/blob/4a47e2369065485dbc8089393b2696defc4d9d82/.circleci/config.yml#L981-L985
The function `push_main_image_set` will remind you of a location where development images get pushed.

Reply:
I went too fast on copying the links from opensource flavor it seems, whoops. I fixed it now. As for those tests that assert image repositories I'm having trouble finding them. Where are they?

Reply:
> Where are they?

Please look for a directory called `testdata` next to `flavor_test.go`.

Regarding the current change to `quay.io/stackrox-io`: this is the same directory where images with community branding and `opensource` flavor go. Currently, `development_build` flavored images have commercial branding. I think that commercial branding shouldn't show up in `quay.io/stackrox-io`.
What are options?
1. Keep images with development_build flavor and commercial branding in quay repo different from `quay.io/stackrox-io`.
2. Remove images with development_build flavor and commercial branding altogether, remove the flavor, clean up the code that pushed them. Use images with `opensource` flavor and community branding for daily development in the engineering team.

(1) is less change and less chance for mistake - you just need to figure out the correct repo to set in this PR.
(2) is more brave, but the simplification may be beneficial.